src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1731,6 +1736,30 @@ void reportPrimaryConstructorParameterShadowing(SimpleNameSyntax node, Symbol sy
             }
         }
 
+#nullable enable
+        /// <summary>
+        /// Report a diagnostic for a 'field' or 'value' identifier that the meaning will
+        /// change when the identifier is considered a contextual keyword.
+        /// </summary>
+        internal void ReportFieldOrValueContextualKeywordConflictIfAny(SyntaxNode syntax, SyntaxToken identifier, BindingDiagnosticBag diagnostics)
+        {
+            string name = identifier.Text;
+            switch (name)
+            {
+                case "field" when ContainingMember() is MethodSymbol { MethodKind: MethodKind.PropertyGet or MethodKind.PropertySet, AssociatedSymbol: PropertySymbol { IsIndexer: false } }:
+                case "value" when ContainingMember() is MethodSymbol { MethodKind: MethodKind.PropertySet or MethodKind.EventAdd or MethodKind.EventRemove }:
+                    {
+                        var requiredVersion = MessageID.IDS_FeatureFieldAndValueKeywords.RequiredVersion();
+                        if (Compilation.LanguageVersion < requiredVersion)
```

Comment:
>if (Compilation.LanguageVersion < requiredVersion) [](http://example.com/codeflow?start=24&length=50)

Would it make sense to check this first, right as we enter the function?

Reply:
Having the check here seems reasonable to me, since we're only checking if the identifier is "name" or "value".